kafka-client/src/main/java/dev/responsive/kafka/internal/db/WriteBatch.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ *  * Copyright 2023 Responsive Computing, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package dev.responsive.kafka.internal.db;
+
+import dev.responsive.kafka.internal.db.partitioning.TablePartitioner;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import dev.responsive.kafka.internal.utils.Result;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+/**
+ * A class for holding and coordinating the pending writes and other metadata involved
+ * in a flush to the remote table
+ */
+public class WriteBatch<K extends Comparable<K>, P> {
```

Comment:
Offline I wasn't sure what the point of this class is if we have `CommitBuffer` but it makes sense to me now - `CommitBuffer` buffers writes and then constructs a `WriteBatch` (which is effectively immutable after creation). If we really want to make that clear we could rename it `ReadyBatch` or something like that to indicate that it's something ready to go instead of something dynamic (🤮 ok, well that name sucks, but you get the point and hoepfully where my confusion came from)